### DISCUSSION TITLE
MediaControlSDK - Steps for saving a jpeg

### DISCUSSION THREAD

**Discussion by cm402816** [AxisCommunications/discussions#154] [Axis Windows SDKs] [answered]
Would someone be able to give me a simple order of operations for storing a jpg image from a camera using the MediaControl SDK. I don't need to see the image in realtime, just save a snapshot to a folder.

Here's what I'm trying now but it doesn't seem to work. I am able to view the stream in the control and save the image when playing, but it won't work with this code.
```
Private Sub btnCaptureJpg_Click(sender As Object, e As EventArgs) Handles btnCaptureJpg.Click
        AMC.MediaURL = "axrtsphttp://192.168.0.1/axis-media/media.amp"
        AMC.MediaType = "h264"
        AMC.MediaUsername = "user"
        AMC.MediaPassword = "password"
        AMC.Play()
        AMC.SaveCurrentImage(0, "C:\temp\image.jpg")
        AMC.Stop()
End Sub
```
let me know if there is a better way I should be doing this.

Thanks!

**Comment by anders-axis** [comment, marked as answer]
Hi,

To just get a snapshot from an Axis product you can just downloading it directly using HTTP with optional parameters:
`http://<camera-ip>/axis-cgi/jpg/image.cgi?resolution=320x240&compression=25&camera=1`
https://www.axis.com/vapix-library/subjects/t10175981/section/t10051110/display (see 'JPEG image (snapshot)')

The code above should typically work, but note that most methods of AMC SDK is asynchronous so you might want to wait a while after the call to Play() before saving the snapshot so that the stream has been fully started. You can also use events to know when the stream has been started, e.g. OnStatusChanged or OnNewImage.

**Reply by cm402816** [reply]
Thanks for the help. It seems like the direct url would work better, but how would I handle connecting to the camera programmatically and passing the username and password without having a pop-up requiring user intervention. The SDK method I tested above seemed to work great in that it handled the authentication for me. Is there a way to do something similar with the URL access?

**Reply by anders-axis** [reply]
Yes, if your only functionality is to download snapshots, then skipping out on the AMC SDK is the way forward.

There are several available HTTP clients you can use to perform this operation programmatically. If you are using .Net for example, there are at least three different clients you can use, for example HttpClient. All of these supports authentication over HTTP and HTTPS (basic and digest).

Found this example how to do this with HttpClient. Hope this helps.
https://stackoverflow.com/questions/10658202/httprequestmessage-and-digest-authentication

**Reply by cm402816** [reply]
Thanks again for the help. I looked at the example you linked and was able to adjust it so that it was working correctly with vb.net. Here's the working code in case anyone else needs it. It grabs an image from the camera via http with username and password and displays it in a picturebox.

```
       Dim strIP As String = txtIP.Text '10.0.0.1

        Dim myURI As String = "http://" & strIP & "/axis-cgi/jpg/image.cgi?"
        Dim Wreq As WebRequest = WebRequest.Create(myURI)
        Wreq.Credentials = New NetworkCredential("username", "password")
        Wreq.Timeout = 5000

        Dim mywebresponse As WebResponse = Wreq.GetResponse()

        Dim stream As Stream = mywebresponse.GetResponseStream

        Dim image As Image = Image.FromStream(stream)

        pb1.Image = image
```
